tendermint/src/validator.rs

```diff
@@ -23,14 +23,20 @@ impl Set {
     /// Create a new validator set.
     /// vals is mutable so it can be sorted by address.
     pub fn new(mut vals: Vec<Info>) -> Set {
-        vals.sort_by(|v1, v2| v1.address.partial_cmp(&v2.address).unwrap());
+        Self::sort_validators(&mut vals);
         Set { validators: vals }
     }
 
     /// Get Info of the underlying validators.
     pub fn validators(&self) -> &Vec<Info> {
         &self.validators
     }
+
+    /// Sort the validators according to the current Tendermint requirements
+    /// (v. 0.33 -> by validator address, ascending)
+    fn sort_validators(vals: &mut Vec<Info>) {
+        vals.sort_by(|v1, v2| v1.address.partial_cmp(&v2.address).unwrap());
```

Comment:
```suggestion
        vals.sort_by_key(|v| v.address);
```

More idiomatic since `account::Id` implements `Ord` and improve type-safety by avoiding the `unwrap` call (which always succeeds because of the `Ord` bound, but could silently start failing if the bound was somehow removed).

Reply:
nice:) thanks for pointing out!